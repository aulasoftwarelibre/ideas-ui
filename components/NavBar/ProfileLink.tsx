import Link from 'next/link';
import { NextRouter, withRouter } from 'next/router';
import React from 'react';

export interface Props {
  path: string;
  label: string;
  router: NextRouter;
}

export const ProfileLink: React.FunctionComponent<Props> = ({
  path,
  label,
  router,
}) => {
  const isActive = router.pathname === path;

  if (isActive) {
    return (
      <span
        className="block px-4 py-2 text-sm leading-5 font-black text-gray-700 hover:bg-gray-100 focus:outline-none focus:bg-gray-100 transition duration-150 ease-in-out"
        role="menuitem"
        data-testid="active-profile-link"
      >
        {label}
      </span>
    );
  }

  return (
    <Link href={path}>
      <a
        className="block px-4 py-2 text-sm leading-5 text-gray-700 hover:bg-gray-100 focus:outline-none focus:bg-gray-100 transition duration-150 ease-in-out"
        role="menuitem"
        data-testid="profile-link"
      >
        {label}
      </a>
    </Link>
  );
};

export default withRouter(ProfileLink);
