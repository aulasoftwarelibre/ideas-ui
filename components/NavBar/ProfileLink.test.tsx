import * as React from 'react';

import { mountMockRouter, renderMockRouter } from '../../utils/test-utils';
import ProfileLink from './ProfileLink';

describe("ProfileLink", () => {
  it("should be rendered correctly", () => {
    const { asFragment } = renderMockRouter(
      <ProfileLink path="/no-active" label="Profile" />,
      {
        router: { pathname: "/active" },
      }
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("should be rendered correctly when current is enabled", () => {
    const { asFragment } = renderMockRouter(
      <ProfileLink path="/active" label="Profile" />,
      {
        router: { pathname: "/active" },
      }
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("should be a link", () => {
    const wrapper = mountMockRouter(
      <ProfileLink path="/active" label="Profile" />,
      {
        router: { pathname: "/" },
      }
    );

    expect(wrapper.find("a")).toHaveLength(1);
  });

  it("should be a span when active", () => {
    const wrapper = mountMockRouter(
      <ProfileLink path="/active" label="Profile" />,
      {
        router: { pathname: "/active" },
      }
    );

    expect(wrapper.find("span")).toHaveLength(1);
  });
});
