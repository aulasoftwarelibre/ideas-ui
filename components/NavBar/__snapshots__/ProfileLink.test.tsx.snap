// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileLink should be rendered correctly 1`] = `
<DocumentFragment>
  <a
    class="block px-4 py-2 text-sm leading-5 text-gray-700 hover:bg-gray-100 focus:outline-none focus:bg-gray-100 transition duration-150 ease-in-out"
    href="/no-active"
    role="menuitem"
  >
    Profile
  </a>
</DocumentFragment>
`;

exports[`ProfileLink should be rendered correctly when current is enabled 1`] = `
<DocumentFragment>
  <span
    class="block px-4 py-2 text-sm leading-5 font-black text-gray-700 hover:bg-gray-100 focus:outline-none focus:bg-gray-100 transition duration-150 ease-in-out"
    role="menuitem"
  >
    Profile
  </span>
</DocumentFragment>
`;
